php/util/class-fieldmanager-util-gutenberg.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Class file for Fieldmanager_Util_Gutenberg
+ *
+ * @package Fieldmanager
+ */
+
+/**
+ * Gutenberg helpers.
+ */
+class Fieldmanager_Util_Gutenberg {
+	public function __construct() {
+		add_filter( 'fm_enqueue_scripts', [ $this, 'add_gutenberg_js_deps' ], 99 );
+
+		// Gutenberg sidebar polyfill.
+		fm_add_script( 'fieldmanager-gutenberg-polyfill', 'js/fieldmanager-gutenberg-support.js' );
```

Comment:
My memory is fuzzy here: Will `fm_add_script()` cause this script to be enqueued? If so, is it now being enqueued on all requests by nature of its being in the constructor?

Reply:
@dlh01 This function triggers `pre_enqueue_script` which checks for `did_action( 'admin_enqueue_scripts' ) || did_action( 'wp_enqueue_scripts' )` which enqueues the script else adds the script handle the to the scripts array and runs `$this->hook_enqueue()` which adds actions for `admin_enqueue_scripts` and `wp_enqueue_scripts` and filters/enqueues there.

Reply:
Right. So, it looks like the script is loading on all requests regardless of whether a field is on the page. It would automatically load on the frontend as well, but there's an `is_admin()` check in `\Fieldmanager_Util_Assets::add_script()` to prevent that from happening.